Add snap interval

nikolai-b opened the pull request:
Also looks like `drop_null_geometries` wasn't being pass through for `geo_list` so I've added that in this PR.

ateucher commented:
Thanks @nikolai-b! I'll probably have time to review this in a week or so, but if you want to check into why the builds are failing that would be awesome. (#23)

nikolai-b commented:
Cheers, will do

nikolai-b commented:
Ok, updated.  I removed 
https://github.com/ateucher/rmapshaper/commit/3ed81c92ac0bddd1fb40c8a811cc992ea16db620
from this pull as that seem to break the specs.  I'm not sure why as it looks like simplify is ignoring dropping null geometries, is that intentional? The docs seem to mention it is always true for spatial objects but not that it is always `FALSE` for `geo_list`s.  Sorry if I missing something obvious.

ateucher commented on R/simplify.R:
```diff
@@ -173,12 +189,13 @@ make_simplify_call <- function(keep, method, keep_shapes, no_repair, snap, explo
   }
 
   if (explode) explode <- "-explode" else explode <- NULL
+  if (snap && !is.null(snap_interval)) snap_interval <- paste("snap-interval=", snap_interval)
```
This should be `paste0` instead of `paste`, but I'll make the fix after I merge it

nikolai-b replied:
Eeek, of course, thanks for spotting.

ateucher commented:
Sorry for taking so long to get to this @nikolai-b  - it looks good! There's one small fix that needs to be made, (see inline comments), but I'll do it after I merge. Then I'll add a few tests.

Thanks so much for the contribution!

As for the dropping null geometries, I went in a circles a bit on that, so I'm not overly surprised it's inconsistent. I'll have a look - would you mind opening a separate issue to remind me?
